### PR TITLE
Xtream Updates and added settings

### PR DESCRIPTION
**Pull request by superolmo**
- Added setting for hiding empty group #207 
- Added setting for hiding adult channels (if marked so by provider) #207 
- Added support for using Hypnotix user agent in Xtream #175 #274 
- Added connection retry in Xtream
- Added progress bar in console for debugging slow processing #272 
- Improved ability to have more than one Xtream provider (untested) #255 
- Improved Xtream loading speed #272 
- Improved logging by using the GUI status bar
- Changed automatic reload from 2 to 4 hours

### CODE REVIEW

File: usr/lib/hypnotix/xtream.py [hunk 1]
```diff
@@ -16,16 +16,17 @@
...
 import json
 import re  # used for URL validation
 import time
-from os import path as osp
 from os import makedirs
+from os import path as osp
+from sys import stdout
 from timeit import default_timer as timer  # Timing xtream json downloads
-from typing import List, Tuple
+from typing import List, Tuple, Protocol
```
Comment (mtwebster) [inline usr/lib/hypnotix/xtream.py:29]: We need to stay compatible with python 3.9 (Debian Bullseye/LMDE5), It doesn't seem as though Protocol is available there

Comment (superolmo) [reply usr/lib/hypnotix/xtream.py:29]: For a moment I though I had something special in my python. Glad it works.

**Comment by mtwebster** [comment]
Sorry for the delay - I'm trying to get an account wit xtream to test things out but they seem to be unresponsive.

**Comment by superolmo** [comment]
No problem, take your time. It is better to get it tested then rushed in.

**Pull request closed without merging**